app/src/main/java/org/sopt/cgv/feature/time/component/SelectableTheatersInModal.kt

```diff
@@ -0,0 +1,206 @@
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import kotlinx.collections.immutable.PersistentList
+import kotlinx.collections.immutable.persistentListOf
+import org.sopt.cgv.R
+import org.sopt.cgv.core.common.extension.noRippleClickable
+import org.sopt.cgv.core.designsystem.theme.CGVTheme
+import org.sopt.cgv.core.designsystem.theme.Gray200
+import org.sopt.cgv.core.designsystem.theme.Gray700
+import org.sopt.cgv.core.designsystem.theme.Gray850
+import org.sopt.cgv.core.designsystem.theme.PrimaryRed400
+import org.sopt.cgv.core.designsystem.theme.White
+
+
+data class MovieTheatersByDetailRegion(
+    val detailRegionName: String,
+    val theaterNames: PersistentList<String>
+)
+
+@Composable
+fun SelectableTheatersInModal(
+    movieTheatersByDetailRegion: PersistentList<MovieTheatersByDetailRegion>,
+    selectedTheaters: MutableState<Set<String>>,
+    modifier: Modifier = Modifier
+) {
+    LazyColumn(
+        modifier = modifier
+            .padding(end = 18.dp)
+    ) {
+        item { Spacer(modifier = Modifier.height(26.dp)) }
+
+        item {
+            Text(
+                text = stringResource(R.string.cgv_theater_selection_guide),
+                style = CGVTheme.typography.small1_l_10,
+                color = Gray700
+            )
+        }
+
+        item { Spacer(modifier = Modifier.height(25.dp)) }
+
+        items(movieTheatersByDetailRegion) { movieTheatersByDetailRegion ->
+            DetailRegionTheaters(
+                detailRegionName = movieTheatersByDetailRegion.detailRegionName,
+                theaterNames = movieTheatersByDetailRegion.theaterNames,
+                selectedTheaters = selectedTheaters
+            )
+        }
+    }
+}
+
+@Composable
+fun DetailRegionTheaters(
+    detailRegionName: String,
+    theaterNames: PersistentList<String>,
+    selectedTheaters: MutableState<Set<String>>,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = Modifier.fillMaxWidth()
+    ) {
+        Text(
+            text = detailRegionName,
+            style = CGVTheme.typography.head0_b_10,
+            color = PrimaryRed400
+        )
+
+        Spacer(modifier = Modifier.height(25.dp))
+
+        theaterNames.forEach {
+            TheaterListItem(
+                theaterName = it,
+                selectedTheaters = selectedTheaters
+            )
+        }
+
+        Spacer(modifier = Modifier.height(36.dp))
+    }
+}
+
+@Composable
+fun TheaterListItem(
+    theaterName: String,
+    selectedTheaters: MutableState<Set<String>>,
```

Comment:
```suggestion
     isSelected: Boolean,
    onTheaterClick: () -> Unit
```
여기서도 이렇게 바꿔주시면

상태 관리와 이벤트 처리가 개선될 수 있을 거 같아요!

1. 상태 관리
기존에` var isSelected by remember { mutableStateOf(false) }`로 컴포넌트 내부에서 상태를 직접 관리하던 것을
`isSelected: Boolean` 파라미터로 상태를 외부에서 전달받아 단방향 데이터 흐름 구현할 수 있습니다!


2. 이벤트 처리
기존에 컴포넌트 내부에서 상태 변경 로직을 직접 처리하신 거 같은데, `onTheaterClick: () -> Unit` 콜백을 통해 클릭 이벤트만 전달하고 실제 처리는 외부로 위임이 가능합니다.

이렇게 하면 컴포넌트는 데이터를 표시하고 이벤트를 전달하는 순수한 UI 표현에 집중할 수 있을 거 같아요!